libfwupdplugin/fu-smbios.c

```diff
@@ -89,6 +93,76 @@ fu_smbios_convert_dt_string (FuSmbios *self, guint8 type, guint8 offset,
 	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
 }
 
+#if defined (__FreeBSD__) || defined (__DragonFly__)
+static void
+fu_smbios_kenv_sysctl_string (FuSmbios *self, guint8 type, guint8 offset,
+				const gchar *buf, gsize bufsz)
+{
+	FuSmbiosItem *item = g_ptr_array_index (self->items, type);
+
+	/* add to strtab */
+	g_ptr_array_add (item->strings, g_strndup (buf, bufsz));
+	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
+}
+
+static void
+fu_smbios_kenv_lookup (const char* sminfo, gchar* buf)
+{
+	g_debug ("DMI request for %s", sminfo);
+	if (kenv (KENV_GET, sminfo, buf, sizeof (buf) - 1) == -1)
```

Comment:
sizeof (buf) isn't going to work, surely? you need to pass `buffsz` into the function as the compiler has no idea how big the char buffer actually is.

Reply:
It worked, but better idea is to pass `bufsz`. Fixed.

Reply:
TIL :)